spec/test-data/translations/file-with-duplicated-keys.strings

```diff
@@ -0,0 +1,66 @@
+/* A comment */
+"key" = "localized translation here";
+/* A value with escaped characters */
+"key.key" = "Hello \"World\"!";
+/* Keys with escaped characters */
+"it's a \" = " = "trap";
+"it's another \\" = "trap";
+/* A multi-line value */
+"error.message" = "One line.
+Another line.";
+/* A multi-line
+ comment
+*/
+"comment" = "comment"
+/* Below are two keys with leading spaces to test against malformatted files */
+ "space" = "localized translation here";
+  "more.space" = "localized translation here";
+/* Below are some entries with unusual spaces between key and value to test against malformed files */
+"nospace"="localized translation here";
+"space.nospace" ="localized translation here";
+"nospace.space"= "localized translation here";
+"lots of spaces"   =  "localized translation here";
+/* Special case: \"%@\" can be tricky to detect */
+"Key escaped parameter \"%@\"" = "localized translation here";
+"Something \"something\" something \"else\"" = "localized translation here";
+/*
+ * Duplicated keys
+ */
+/* Consecutive duplicated keys */
+"dup1" = "localized translation here";
+"dup1" = "localized translation here";
+/* Duplicated keys with other entries in between */
+"dup2" = "localized translation here";
+"not dup" = "localized translation here";
+"dup2" = "localized translation here";
+"dup3" = "localized translation here";
+"not dup 2" = "localized translation here";
+"not dup 3" = "localized translation here";
+"dup3" = "localized translation here";
+/* Duplicated keys with different translations are still duplicated */
+"dup4" = "a translation";
+"dup4" = "a different translation";
+/* Duplicated comments should be ignored */
+"key with dup comment" = "localized translation here";
+/* Duplicated comments should be ignored */
+"different key with dup comment" = "localized translation here";
+/* Unicode codepoint escape — see https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/LoadingResources/Strings/Strings.html#//apple_ref/doc/uid/10000051i-CH6-SW13 */
+/* Consecutive... */
+"\U0025 key" = "\U0025 is the % symbol";
+"\U0025 key" = "\U0025 is the % symbol";
+/* ...and not consecutive */
+"\U0026 key" = "\U0025 is the & symbol";
+"unicode\U0020key" = "\U0020 is the space character";
+"\U0026 key" = "\U0025 is the & symbol";
+/* Special case: \"%@\" can be tricky to detect */
+"The Google account \"%@\" doesn't match any account on WordPress.com" = "localized translation here";
+/* A red herring that might be seen as duplicate to the previous one if our RegExp does not correctly account for escaped quotes */
+"The Google account \"%@\" is invalid" = "localized translation here";
```

Comment:
This passes because the RegEx actually bases its detection of the end of the range containing the key… based on `=` (the `[^*]+` is kinda pointless given you have a `.*` right after it, so the real thing that makes it stop at end of key is the `=` in your RegEx after all)

Which means that while this red herring passes your test, the same line but with also some addition `=` somewhere in the line (be it in the key, the translation, or as a `/* comment */` at the end of the line) would still throw you off rails.

Reply:
In [`6478d58`](https://github.com/wordpress-mobile/release-toolkit/pull/342/commits/6478d58557f477bee78278b0c2e42498de5c4889) I added more tests inputs based on my understanding of your [suggestion](https://github.com/wordpress-mobile/release-toolkit/pull/342/#discussion_r816760766) and the observation above: 

> It also won't work with keys that contain \" in their name — because of the [^"]+ thing. The .* that follows might accidentally workaround this, actually rendering the preceding [^"]+ kinda useless, but that also risks making the .* be greedy (e.g. "Login to a \"%@\" account" = "Login to a \"%@\" account"; // Here %@ = the service name, like Google)

The tests passed on my machine. I need to jump offline so I don't know if they'll pass on CI, too, but I don't see why not (_last famous words_).

If I misunderstood your suggestion, could you suggest more inputs for `file-with-duplicated-keys.strings` so we can get the tests failing and go from there? Thanks.